xblock/field_data.py

```diff
@@ -145,17 +144,51 @@ def __init__(self, scope_mappings):
         """
         self._scope_mappings = scope_mappings
 
-    def _field_data(self, block, name):
-        """Return the field data for the field `name` on the :class:`~xblock.core.XBlock` `block`"""
-        scope = block.fields[name].scope
-
+    def _find_scope_in_mapping(self, scope):
+        """Summary
+        
+        Args:
+            scope (TYPE): Description
+        
+        Returns:
+            TYPE: Description
+        """
         if scope not in self._scope_mappings:
             raise InvalidScopeError(scope)
 
         return self._scope_mappings[scope]
 
+    def _field_data(self, block, name):
+        """Return the field data for the field `name` on the :class:`~xblock.core.XBlock` `block`"""
+        scope = block.fields[name].scope
+
+        return self._find_scope_in_mapping(scope)
+
+    def _shared_field_data(self, remote_scope):
+        """Summary
+        
+        Args:
+            remote_scope (TYPE): Description
+        
+        Returns:
+            TYPE: Description
+        
+        Deleted Parameters:
+            block (TYPE): Description
+            name (TYPE): Description
+        """
+        return self._find_scope_in_mapping(remote_scope)
+
     def get(self, block, name):
-        return self._field_data(block, name).get(block, name)
```

Comment:
I am re-considering the part of code changes,  I am trying to keep original XBlock code intact and push more changes to the platform side, but still trying to find out if that is feasible.